src/helpers/s3.js

```diff
@@ -0,0 +1,80 @@
+const s3 = require('s3');
+const AWS = require('aws-sdk');
+
+/**
+ * Upload dir to S3
+ * @param {object} credentials - AWS credentials
+ * @param {string} bucket - S3 bucket name
+ * @param {string} dirPath - source dir path
+ */
+function upload(env, credentials, bucket, dirPath, getClient){
+  return new Promise((resolve, reject) => {
+    const client = getS3Client(credentials);
+    const uploader = client.uploadDir({
+      localDir: dirPath,
+      deleteRemoved: true,
+      s3Params: {
+        Bucket: bucket,
+        Prefix: env
+      }
+    });
+    uploader
+      .on('err', reject)
+      .on('end', resolve);
+    if(typeof getClient === 'function'){
+      getClient(uploader);
+    }
+  });
+}
+
+/**
+ * Download dir from S3
+ * @param {object} credentials - AWS credentials
+ * @param {string} bucket - S3 bucket name
+ * @param {string} dirPath - destination dir path
+ */
+function download(env, credentials, bucket, dirPath, getClient){
+  return new Promise((resolve, reject) => {
+    const client = getS3Client(credentials);
+    const uploader = client.downloadDir({
+      localDir: dirPath,
+      deleteRemoved: true,
+      s3Params: {
+        Bucket: bucket,
+        Prefix: env
+      }
+    });
+    uploader
+      .on('err', reject)
+      .on('end', resolve);
+    if(typeof getClient === 'function'){
+      getClient(uploader);
+    }
+  });
+}
+
+/**
+ * Return S3 client
+ * @param {object} credentials - AWS credentials, format:
+ * {
+ *  accessKeyId: {{ID}},
+ *  secretAccessKey: {{SECRET}},
+ *  region: 'us-west-2'
+ * }
+ */
+function getS3Client(credentials){
```

Comment:
It would be good idea to check here if user passed credentials in env variables and make some meaningful error here